modules/develop/pages/data-transforms/build.adoc

```diff
@@ -121,25 +199,19 @@ import (
 )
 
 func main() {
-    transform.OnRecordWritten(handleRecord)
+    transform.OnRecordWritten(myTransform)
 }
 
-func handleRecord(event transform.WriteEvent, writer transform.RecordWriter) error {
+func myTransform(event transform.WriteEvent, writer transform.RecordWriter) error {
   record := event.Record()
   if record.Key == nil {
     // Log the error and continue processing
     log.Println("Error: Record key is nil")
-    // Do not return the error
+    // Skip this record
     return nil
   }
 
-  err := writer.Write(record)
-  if err != nil {
-    // Log the error and continue processing
-    log.Println("Error writing record:", err)
-  }
-  // Ensure no error is returned
-  return nil
+  return writer.Write(record)
 }
```

Comment:
idle thought: broadly this seems feels like a complete example, exhibiting both error-handling behaviors. If we detect a "bad" record, it can be discarded outright and processing continues. If writing a "good" record _fails_, we allow that error to propagate.

Reply:
Thanks! I've cleaned this part up to include this single complete example.